### PR TITLE
fix(Model): Apply codegen changes test models

### PR DESCRIPTION
**Pull request by lawmicha**
*Issue #, if available:*
codegen issue: https://github.com/aws-amplify/amplify-codegen/issues/104
iOS issue: https://github.com/aws-amplify/amplify-ios/issues/1116

*Description of changes:*
This applies the proposed codegen changes from this PR: https://github.com/aws-amplify/amplify-codegen/pull/142 
doc updates: https://github.com/aws-amplify/docs/pull/3306

*Testing done:*
- [x] API integ tests related to the model changes
- [x] DataStore integ tests related to the model changes

*Check points:*

- [ ] Added new tests to cover change, if needed
- [ ] All unit tests pass
- [ ] All integration tests pass
- [ ] Documentation update for the change if required
- [ ] PR title conforms to conventional commit style

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

**Review by palpatim — APPROVED** [review]

**Comment by lawmicha** [comment]
this has been tested with the latest generated code from the codegen PR at this time: https://github.com/aws-amplify/amplify-codegen/pull/142#pullrequestreview-686423369

File: AmplifyPlugins/DataStore/AWSDataStoreCategoryPluginIntegrationTests/TestSupport/TestModelRegistration.swift [hunk 1]
```diff
@@ -32,7 +32,7 @@ struct TestModelRegistration: AmplifyModelRegistration {
         registry.register(modelType: ListStringContainer.self)
         registry.register(modelType: EnumTestModel.self)
         registry.register(modelType: NestedTypeTestModel.self)
-        registry.register(modelType: CustomerOrder.self)
+        //registry.register(modelType: CustomerOrder.self)
```
Comment (palpatim) [inline AmplifyPlugins/DataStore/AWSDataStoreCategoryPluginIntegrationTests/TestSupport/TestModelRegistration.swift:35]: nit: remove

**Review by palpatim — APPROVED** [review]
minor comment, otherwise LGTM

File: AmplifyPlugins/DataStore/AWSDataStoreCategoryPluginIntegrationTests/TestSupport/TestModelRegistration.swift (hunk 1 above)
Comment (lawmicha) [inline AmplifyPlugins/DataStore/AWSDataStoreCategoryPluginIntegrationTests/TestSupport/TestModelRegistration.swift:35]:
```suggestion
        registry.register(modelType: CustomerOrder.self)
```

**Pull request merged**